src/lib.rs

```diff
@@ -60,6 +71,62 @@ pub fn parse(contents: &str) -> Result<EditorConfigINIAST, Error<Rule>> {
 							}
 						}
 					}
+					Rule::comment => {
+						let mut inner_rules = line.into_inner();
+						let comment = Item::Comment(Comment {
+							indicator: inner_rules.next().unwrap().as_str().to_string(),
+							value: inner_rules.next().unwrap().as_str().to_string(),
+							raws: Raws {
+								before: Some("".to_string()),
+								after: Some("".to_string()),
+								newline: Some(inner_rules.next().unwrap().as_str().to_string()),
+							},
+						});
+						match &mut current_section {
```

Comment:
This matching gets repeated throughout this file, so I'm thinking of just having a single body that's always there so that it cleans this up. Seeing as the ast root and each section has a body, I don't see why this would be a problem, but @florianb you didn't like something about this idea in the past.

Reply:
I fear i do not understand but feel welcome to try it. I was in general aiming for a structure where each element consists of the same parent/child-structure to allow recursive usage.